project/frontend/src/components/LocationFinder.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+class LocationFinder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      locationString: '',
+    };
+    this.changeState = this.changeState.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  changeState(event) {
+    this.setState({ locationString: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.getLocation(this.state.locationString);
+  }
+
+  getLocation(userInput) {
```

Comment:
Since some vars (proxyUrl, headers) are common between getLocation here and getRecommendation, I am thinking of moving those functions to the same file and having file level vars for these that can be used

Reply:
I think you should use the Geocoder JavaScript API instead of this herokuapp proxy to get around the CORS errors. You can read about it here https://developers.google.com/maps/documentation/javascript/geocoding. It seems really simple to use.